File: spring-data-requery/src/test/java/org/springframework/data/requery/repository/query/DefaultRequeryEntityMetadataTest.java
package org.springframework.data.requery.repository.query;

import org.assertj.core.api.Assertions;
import org.junit.Test;
import org.springframework.data.requery.domain.basic.AbstractBasicUser;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * org.springframework.data.requery.repository.query.DefaultRequeryEntityMetadataTest
 *
 * @author debop
 * @since 18. 6. 8
 */
public class DefaultRequeryEntityMetadataTest {

    @Test
    public void rejects_null_domainType() {
        Assertions.assertThatThrownBy(() -> {

            new DefaultRequeryEntityMetadata(null);

        }).isInstanceOf(IllegalArgumentException.class);
    }

    @Test
    public void returnsConfiguredType() {
        DefaultRequeryEntityMetadata<Foo> metadata = new DefaultRequeryEntityMetadata<>(Foo.class);
        assertThat(metadata.getJavaType()).isEqualTo(Foo.class);
    }

    @Test
    public void returnSimpleClassNameAsEntityNameByDefault() {
        DefaultRequeryEntityMetadata<Foo> metadata = new DefaultRequeryEntityMetadata<>(Foo.class);
        assertThat(metadata.getEntityName()).isEqualTo(Foo.class.getSimpleName());
    }

    @Test
    public void returnCustomizedEntityNameIfConfigured() {
        DefaultRequeryEntityMetadata<AbstractBasicUser> metadata = new DefaultRequeryEntityMetadata<>(AbstractBasicUser.class);
        assertThat(metadata.getEntityName()).isEqualTo("BasicUser");
    }


    static class Foo {}

}

File: spring-data-requery/src/test/kotlin/org/springframework/data/requery/repository/query/DefaultRequeryEntityMetadataTest.kt
package org.springframework.data.requery.repository.query

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.springframework.data.requery.domain.sample.AbstractUser
import org.springframework.data.requery.domain.sample.User

/**
 * org.springframework.data.requery.repository.query.DefaultRequeryEntityMetadataTest
 *
 * @author debop
 */
class DefaultRequeryEntityMetadataTest {

    class Foo {}

    @Test
    fun `return configured type`() {

        val metadata = DefaultRequeryEntityMetadata(Foo::class.java)
        assertThat(metadata.javaType).isEqualTo(Foo::class.java)
    }

    @Test
    fun `return simple class name as entity name by default`() {

        val metadata = DefaultRequeryEntityMetadata(Foo::class.java)
        assertThat(metadata.entityName).isEqualTo(Foo::class.java.simpleName)
    }

    @Test
    fun `return customized entity name if configured`() {

        val metadata = DefaultRequeryEntityMetadata(AbstractUser::class.java)
        assertThat(metadata.entityName).isEqualTo("User")

        val metadata2 = DefaultRequeryEntityMetadata(User::class.java)
        assertThat(metadata2.entityName).isEqualTo("User")
    }
}